django-backend/fecfiler/web_services/summary/summary.py

```diff
@@ -2,21 +2,70 @@
 from fecfiler.transactions.models import Transaction
 from django.db.models import Q, Sum
 from django.db.models.functions import Coalesce
+import logging
+
+logger = logging.getLogger(__name__)
 
 
 class SummaryService:
     def __init__(self, report) -> None:
         self.report = report
 
     def calculate_summary(self):
+        summary = {
+            "a": self.calculate_summary_column_a(),
+            "b": self.calculate_summary_column_b()
+        }
+
+        return summary
+
+    def calculate_summary_column_a(self):
         report_transactions = Transaction.objects.filter(report=self.report)
-        # just line 15
-        sa15_query = Q(~Q(memo_code=True), form_type="SA15")
-        line_15 = self._create_contribution_sum(sa15_query)
         # line 12
         sa12_query = Q(~Q(memo_code=True), form_type="SA12")
         line_12 = self._create_contribution_sum(sa12_query)
-        summary = report_transactions.aggregate(line_15=line_15, line_12=line_12)
+        # line 15
+        sa15_query = Q(~Q(memo_code=True), form_type="SA15")
+        line_15 = self._create_contribution_sum(sa15_query)
+        # line 17
+        sa17_query = Q(~Q(memo_code=True), form_type="SA17")
+        line_17 = self._create_contribution_sum(sa17_query)
+        # line 37 (intentional duplicate of Line 15)
+        line_37 = line_15
+
+        # build summary
+        summary = report_transactions.aggregate(
+            line_12=line_12,
+            line_15=line_15,
+            line_17=line_17,
+            line_37=line_37
+        )
+        return summary
+
+    def calculate_summary_column_b(self):
+        committee = self.report.committee_account
+        report_date = self.report.coverage_from_date
```

Comment:
you're going to want this to be `report.coverage_through_date` so it includes transactions within the report
below, transactions within the report are not counted:
<img width="1145" alt="image" src="https://user-images.githubusercontent.com/97105825/224363667-a3413fb0-92d3-4735-9163-95cb46650bf2.png">

Reply:
and using `date_lt` so it's an exclusive boundary

Reply:
Ah right, good catch